README.md

```diff
@@ -1,2 +1,246 @@
+- **Schema development and application in migration**
+- **Create User Account, Categories and Transactions**
+- **Employs Devise and Cancancan for authentication and authorization respectively**
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project you need:
+
+- Git (for running git related commands only)
+- An IDE (e.g. VS Code) - For viewing and accessing the files only
+- A system with a good OS like Mac Os, Linux or Windows etc
+- PostgreSql (https://www.postgresql.org/download/)
+- Ruby on Rails (https://gorails.com/setup/windows/10)
+
+### Setup
+
+- Depending on your OS, download and install PostgreSql:
+  - You may watch this video (https://www.youtube.com/watch?v=KuQUNHCeKCk)
+- Open your code editor (e.g. VS Code).
+- At the terminal, clone this repository to your desired directory.
+- Make sure the PostgreSql server is running
+  - You may refer to https://www.postgresql.org/docs/current/tutorial-start.html
+- Download and install Ruby on Rails
+  - Following the instructions at https://gorails.com/setup/windows/10
+
+- Use `git clone` to get your local copy of the project.
+  >> At the prompt (bash) enter:
+  ```
+  git clone https://github.com/ometman/budget-app.git
+  ```
+
+### Install Packages and Run Tests <a name="install"></a>
+
+ Run `npm install` and `bundle install` to set up the required packages and gems.
+
+```
+bash > npm install
+````
+```
+bash > bundle install
+````
+ Run Tests <a name="run-tests"></a>
+  To run tests run the following command in your terminal
+    ```
+    Rspec
+    ```
+
+ Run Linters
+  - You can also check linter errors
+  - Make sure Rubocop and stylelint are installed
+     Install Rubocop, run the following command:
+    ```
+    gem install --no-document rubocop -v '>= 1.0, < 2.0'
+    ```
+    Install Stylelint, run the following command:
+    ```
+    npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
+    ```
+  - Now, run the following:
+    ```
+    npx stylelint "**/*.{css,scss}"
+    ```
+
+### Usage <a name="usage"></a>
+
+To run the project locally, execute the following command:
+
+- Start and login to the database server (for Windows): 
+  ```
+  run psql
+  ```
+  Or start PgAdmin
+
+- Launch the app with the following command to start the rails server:
+  ``
+  rails s
+ ``
+ 
+- Voila! Now, go to the browser and enter: localhost:3000 or 127.0.0.1:3000
+
+### Deployment
+
+You can deploy this project using:
+
+- GitHub Pages from a specific branch (e.g. main).
+- You can also used any cloud service like gcloud or Azure.
+- A deployed version can be found here:
+ [https://proper-rivalee-microverse-student.koyeb.app/]
```

Comment:
- [ ] Please kindly know that your deployed project has an error. please fix:

<img width="700" alt="image" src="https://github.com/ometman/budget-app/assets/52934128/e019aaf0-b2ec-4c96-9574-14c21bf7f92d">